src/matrix_multiply_add.jl

```diff
@@ -390,11 +390,11 @@ end
 
 function combine_products(expr_list)
     filtered = filter(expr_list) do expr
-        if expr.head != :call || expr.args[1] != :*
+        if expr.head != :call || expr.args[1] !== :*
```

Comment:
Maybe it should also be `expr.head !== :call`?

Reply:
As long as all the elements in `expr_list` are known by inference to be `Expr`s, it's not necessary:

```julia
julia> @code_typed (ex->ex.head)(:(sin(3.2)))
CodeInfo(
1 ─ %1 = Base.getfield(ex, :head)::Symbol
└──      return %1
) => Symbol
```

I typically use `===` and `!==` only when it makes a difference (when one of the two sides is uninferred).